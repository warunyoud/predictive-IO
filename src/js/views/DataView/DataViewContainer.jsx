import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';

import Dropzone from 'react-dropzone';
import { runPythonScript } from 'scripts/python-interface';

class DataViewContainer extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      accept: '',
      files: [],
      dropzoneActive: false,
      loadingMessage: ''
    }
  }

  onDragEnter() {
    this.setState({
      dropzoneActive: true
    });
  }

  onDragLeave() {
    this.setState({
      dropzoneActive: false
    });
  }

  onDrop(files) {
    const args = [
      `${process.env.workspacePath}/dataset`,
      _.map(files, (file) => file.path)
    ]
    runPythonScript('src/python/add_dataset.py', args, this.props.onDataRecieved, this.props.onProcessFinished, 'text');
    this.props.onProcessStarted();
    this.setState({
      files,
      dropzoneActive: false,
    });
  }

  render() {
    const overlayStyle = {
      color: 'rgba(0,0,0,0.15)',
      border: 'solid',
    };
    return (
      <div>
        <h1>Input a Data</h1>
        <h2>Adding Files</h2>
        <Dropzone
          className='dropzone'
          style={this.state.dropzoneActive ? overlayStyle : null}
          accept={this.state.accept}
          onDrop={this.onDrop.bind(this)}
          onDragEnter={this.onDragEnter.bind(this)}
          onDragLeave={this.onDragLeave.bind(this)}
          >
          <div style={{position: 'absolute', top: '35%', width: '100%', textAlign: 'center'}}>
            <i className='material-icons dropzone-icon'>file_upload</i>
            <h2>Drop files here or click to upload.</h2>
          </div>
        </Dropzone>
      </div>

    );
  }
}

DataViewContainer.propTypes = {
  onProcessStarted: PropTypes.func.isRequired,
  onDataRecieved: PropTypes.func.isRequired,
  onProcessFinished: PropTypes.func.isRequired
}

export default DataViewContainer;
