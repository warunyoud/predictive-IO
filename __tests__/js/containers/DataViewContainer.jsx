import React from 'react';
import DataViewContainer from '../../../src/js/containers/DataViewContainer';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<DataViewContainer
      onProcessStarted={ jest.fn() }
      onDataRecieved={ jest.fn() }
      onProcessFinished={ jest.fn() }
      />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
