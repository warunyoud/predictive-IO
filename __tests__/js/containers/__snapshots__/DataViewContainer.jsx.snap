// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1>
    Input a Data
  </h1>
  <h2>
    Adding Files
  </h2>
  <div
    aria-disabled={false}
    className="dropzone"
    onClick={[Function]}
    onDragEnter={[Function]}
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      style={
        Object {
          "position": "absolute",
          "textAlign": "center",
          "top": "35%",
          "width": "100%",
        }
      }
    >
      <i
        className="material-icons dropzone-icon"
      >
        file_upload
      </i>
      <h2>
        Drop files here or click to upload.
      </h2>
    </div>
    <input
      accept=""
      autoComplete="off"
      disabled={false}
      multiple={true}
      onChange={[Function]}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "opacity": 0.00001,
          "pointerEvents": "none",
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      type="file"
    />
  </div>
</div>
`;
